### ISSUE TITLE
React portlet created via Yeoman gives Liferay AMD Loader error

### ISSUE THREAD

**Issue by sandwichnyc** [liferay/liferay-js-toolkit#374]
Environment:
Liferay DXP 7.1 SP2, FP10
JS portlet extender 1.1.0

Hi - I posted this on the Liferay dev forum but figured I'd try here too.

https://liferay.dev/forums/-/message_boards/message/114410922

I'm still unable to get a React portlet generated via the Yeoman liferay-js generator to work in my portal.  `npm run deploy` seems to deploy to the server correctly, however when I add the portlet to a page I get this error in the browser:

`Liferay AMD Loader: Unhandled failure: <script src=​"http:​/​/​localhost:​8080test-reactjs@1.0.0.js?languageId=en_US">​</script>​ while resolving modules: ["test-reactjs@1.0.0"]`

(Note that it's missing a slash after locahost:8080)

In addition I get this error in the logs:

```
2019-07-29 16:54:25.066 ERROR [Framework Event Dispatcher: Equinox Container: 22946b6e-ed24-4806-9968-496b1929990a][com_liferay_frontend_js_loader_modules_extender:97] FrameworkEvent ERROR 
java.lang.NullPointerException
	at com.github.yuchi.semver.Version.compareTo(Version.java:158)
	at com.github.yuchi.semver.Version.compareTo(Version.java:5)
	at java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:469)
	at java.util.Collections$ReverseComparator2.compare(Collections.java:5178)
	at java.util.TimSort.binarySort(TimSort.java:296)
	at java.util.TimSort.sort(TimSort.java:239)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1462)
	at com.liferay.frontend.js.loader.modules.extender.internal.npm.NPMRegistryImpl._refreshJSModuleCaches(NPMRegistryImpl.java:341)
	at com.liferay.frontend.js.loader.modules.extender.internal.npm.NPMRegistryImpl._processBundle(NPMRegistryImpl.java:279)
	at com.liferay.frontend.js.loader.modules.extender.internal.npm.NPMRegistryImpl.access$400(NPMRegistryImpl.java:71)
	at com.liferay.frontend.js.loader.modules.extender.internal.npm.NPMRegistryImpl$NPMRegistryBundleTrackerCustomizer.addingBundle(NPMRegistryImpl.java:428)
	at com.liferay.frontend.js.loader.modules.extender.internal.npm.NPMRegistryImpl$NPMRegistryBundleTrackerCustomizer.addingBundle(NPMRegistryImpl.java:423)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:475)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:450)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:908)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:230)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:137)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:129)
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:191)
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:476)
	at org.eclipse.osgi.container.Module.start(Module.java:467)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:428)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1258)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1230)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:512)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:361)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:312)
```

Hoping that someone can assist.  As I mentioned in the forum thread, this may have started after I added/removed a package that had a dependency on react-scripts but I don't know if that was the culprit.

**Comment by naderjafari** [comment]
Hi @sandwichnyc 
We have similar issue
This issue resolved for me by updating a dependency in package.json
please attach your package.json

**Comment by sandwichnyc** [comment]
Here's my package.json:

```
{
	"name": "test-reactjs",
	"version": "1.0.0",
	"description": "Test Reactjs",
	"devDependencies": {
		"liferay-npm-bundler": "^2.10.0",
		"liferay-npm-build-support": "^2.10.0",
		"copy-webpack-plugin": "4.6.0",
		"webpack": "4.29.6",
		"webpack-cli": "3.3.0",
		"webpack-dev-server": "3.2.1",
		"babel-cli": "6.26.0",
		"babel-preset-env": "1.7.0",
		"babel-preset-react": "6.24.1",
		"babel-loader": "7.1.5"
	},
	"dependencies": {
		"react": "16.8.6",
		"react-dom": "16.8.6"
	},
	"scripts": {
		"build": "babel --source-maps -d build src && npm run copy-assets && liferay-npm-bundler",
		"copy-assets": "lnbs-copy-assets",
		"deploy": "npm run build && lnbs-deploy",
		"start": "lnbs-start"
	},
	"portlet": {
		"com.liferay.portlet.display-category": "category.sample",
		"com.liferay.portlet.header-portlet-css": "/css/styles.css",
		"com.liferay.portlet.instanceable": true,
		"javax.portlet.name": "test_reactjs",
		"javax.portlet.security-role-ref": "power-user,user",
		"javax.portlet.display-name": "Test Reactjs"
	},
	"main": "index.js"
}

```

I believe it's the version that was generated out of the box with the Yeoman generator.  I don't think I've added any packages since then.

**Comment by izaera** [comment]
If you add and remove a package it's good to erase node_modules and run npm install again when troubleshooting. This is because you may have stale packages around (happened to me lots of times).

I would also advice to clean everything you can (output or any modified files) until you get to the point where it worked and then continue adding things step by step until you reproduce the error so as to know what is exactly causing the issue.

Also, make sure to have the latest versions for every liferay-npm-* package.

**Comment by sandwichnyc** [comment]
HI @izaera - I did finally get this working again, although I tried a few things so I can't pinpoint what exactly worked.  It might have been a combination of getting the latest version of the bundler and setting `javascript.fast.load=false` in the ROOT/WEB-INF/classes portal-developer.properties file (I had it set in the override version of this file in the server root, but maybe it wasn't actually being overridden. 

Thanks again for your help.
